csharp/send-http-request/Index.cs

```diff
@@ -0,0 +1,86 @@
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+static readonly HttpClient http = new HttpClient();
+
+public async Task<RuntimeResponse> Main(RuntimeRequest req, RuntimeResponse res)
+{
+    string url = null;
+    string method = null;
+    string body = null;
+    string contentType = null;
+    Dictionary<string,string> headers = null;
+    if (!string.IsNullOrEmpty(req.Payload))
+    {
+        var payload = JsonConvert.DeserializeObject<Dictionary<string, object>>(req.Payload, settings: null);
+        url = !payload.ContainsKey("url") ? null : (string)payload["url"];
+        method = !payload.ContainsKey("method") ? null : (string)payload["method"];
+        contentType = !payload.ContainsKey("content_type") ? null : (string)payload["content_type"];
+        body = !payload.ContainsKey("body") ? null : payload["body"].ToString();
+        headers = JsonConvert.DeserializeObject<Dictionary<string, string>>(payload.ContainsKey("headers") ? payload["headers"].ToString() : "");
+    }
+
+    // Check for input variable correction
+    if (url == null || method == null)
+    {
+        return res.Json(new()
+        {
+            { "success", false },
+            { "message", "Custom data is in incorrect format or don't exist" }
+        });
+    }
+
+    if (!method.Equals("GET") && !method.Equals("POST") && !method.Equals("PUT") &&
+        !method.Equals("DELETE") && !method.Equals("HEAD") && !method.Equals("OPTIONS") &&
+        !method.Equals("TRACE"))
+    {
+        return res.Json(new()
+            {
+                { "success", false },
+                { "message", "HTTP method is in inccorect format" }
+            });
+    }
+
+    var request = (HttpWebRequest)WebRequest.Create(url);
+    request.Method = method;
+
+    if (headers != null)
+    {
+        foreach(string key in headers.Keys)
+        {
+            request.Headers.Add(key, headers[key]);
+        }
+    }
+
+    if(body != null)
+    {
+        request.ContentType = contentType;
+        var streamWriter = new StreamWriter(request.GetRequestStream());
+        streamWriter.Write(body);
+    }
```

Comment:
Can we please add additional checks in the `if` statement to ensure that if it is a `GET` Request (or other types that don't have a request body), then the `body` and `content type` remain `null`?

In order to debug the empty responses from the functions, I moved the `try` block entry further up and received the response: `Cannot send a content-body with this verb-type.`

![image](https://user-images.githubusercontent.com/31401437/198997410-2b807e53-5e24-4af7-a806-74efafa1beff.png)

Removing the `body` and `content_type` properties from the payload solved this issue. However, keeping the `body` as an empty string triggers the error.

Reply:
```suggestion
    if(!String.IsNullOrEmpty(body))
    {
        request.ContentType = contentType;
        var streamWriter = new StreamWriter(request.GetRequestStream());
        streamWriter.Write(body);
    })
    {
        request.ContentType = contentType;
        var streamWriter = new StreamWriter(request.GetRequestStream());
        streamWriter.Write(body);
    }
```

This should do the trick.